crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,479 @@
+use crate::{utils::is_node_equal, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{self, js_binary_expression, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsStatement, JsBinaryExpression, JsBinaryOperator,
+    JsCallExpression, JsIfStatement, JsLanguage, JsLogicalExpression, JsLogicalOperator,
+    JsParenthesizedExpression, JsSyntaxKind, JsUnaryOperator, JsWhileStatement, JsYieldArgument,
+    JsYieldExpression, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, SyntaxTriviaPiece, TriviaPieceKind, WalkEvent};
+
+declare_rule! {
+    /// Disallow the use of yoda expressions.
+    ///
+    /// A Yoda expression is a programming style where the two parts of the expression are reversed from the typical order in a conditional statement.
+    /// A Yoda expression places the constant portion of the expression on the left side of the conditional statement. They can be confusing to some people,
+    /// the rule forbids the use of it to improve code readability.
+    ///
+    /// The rule have an exception for range expressions, expressions like `0 < value && value < 1` or `value <= 0 || 1 < value` are allowed.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if ("red" == value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (true === value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (5 != value) {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (value === "red") {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value === value) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value != 5) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (0 < value && value < 1) {}
+    /// ```
+    pub NoYodaExpression {
+        version: "next",
+        name: "noYodaExpression",
+        language: "js",
+        sources: &[RuleSource::Eslint("yoda")],
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for NoYodaExpression {
+    type Query = Ast<JsBinaryExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let left = node.left().ok();
+        let right = node.right().ok();
+        let operator = node.operator().ok();
```

Comment:
Inside rules, if you return an `Option` inside the `run` function, always use `.ok()?` when extracting/querying nodes. There's no need to keep the `Option` around unless you're doing an existing check

Reply:
I wasn't familiar with the `?` operator until now and... damn, it really makes things a lot easier 👀, I tried to use it on all the possible places